lib/opal/nodes/top.rb

```diff
@@ -108,6 +180,34 @@ def compile_end_construct
       def version_comment
         "/* Generated by Opal #{Opal::VERSION} */"
       end
+
+      private
+
+      def generate_import_name(module_name)
+        # generate random import name for a ruby module_name. Also replaces some characters that are illegal in JS import names.
+        module_name.gsub('.', 'o_').gsub('-', '_').gsub('/', '_').gsub('@', '_at_') + rand(36**8).to_s(36)
+      end
+
+      def import_child_paths(import_lines, base_dir, module_path)
+        # recursively walk a directory and generate import lines for all .rb/.js files
+        directory_path = base_dir + module_path
+        directory_path.each_child do |child_path|
+          if child_path.directory?
+            import_child_paths(import_lines, base_dir, child_path.expand_path.to_s[(base_dir.expand_path.to_s.length+1)..-1])
+          elsif child_path.file?
+            path_s = child_path.basename.to_s
+            if path_s.end_with?('.rb') || path_s.end_with?('.js')
+              module_path = child_path.expand_path.to_s[(base_dir.expand_path.to_s.length+1)..-4]
```

Comment:
`-4` is a magical constant

Reply:
all requested changes are meanwhile marked outdated. So i resolve this conversion for now.
Ill work on that branch soon again, merging master, etc.